### PR TITLE
Allowed stdlibs

### PR DESCRIPTION
**Pull request by myronmarston**
RSpec-core's changes for rspec/rspec-support#179.

/cc @JonRowe

### CODE REVIEW

File: spec/rspec/core_spec.rb [hunk 1]
```diff
@@ -1,11 +1,55 @@
...
+      'require "rspec/core"',
+      # Prevent rspec/autorun from trying to run RSpec.
+      'RSpec::Core::Runner.disable_autorun!'
+    ], :skip_spec_files => %r{/fake_libs/},
+    :allowed_loaded_feature_regexps => [
+      /optparse\.rb/,   # Used by OptionParser.
+      /set\.rb/,        # used in a few places but being removed in #1870.
+      /rbconfig\.rb/,   # loaded by rspec-support for OS detection.
+      /shellwords\.rb/, # used by ConfigurationOptions and RakeTask.
+      /stringio/,       # Used by BaseFormatter.
```
Comment (JonRowe) [inline spec/rspec/core_spec.rb]: You know we could probably create something to replace this too...

File: spec/rspec/core_spec.rb [hunk 2]
```diff
@@ -1,11 +1,55 @@
...
+      #     dependencies.
+      #   * We're really only concerned with these issues w.r.t. rspec-mocks and
+      #     rspec-expectations from within their spec suites. Here we care only about
+      #     rspec-core, so avoiding loading them helps keep the spec suites independent.
+      #   * These are some of the slowest specs we have, and cutting out the loading
+      #     of external libraries cuts down on how long these specs take.
+      #
+      # To facilitate the avoidance of loading certain libraries, we have a bunch
+      # of files in `support/fake_libs` that substitute for the real things when
+      # we put that directory on the load path. Here's the list:
```
Comment (JonRowe) [inline spec/rspec/core_spec.rb]: Nice...

File: spec/rspec/core_spec.rb (hunk 1 above)
Comment (myronmarston) [inline spec/rspec/core_spec.rb]: Yep, I was thinking the same, but figure we can circle back to it.  I'd rather get this merged and figure that ou tlater.

**Comment by myronmarston** [comment]
This is ready for review. /cc @JonRowe

**Comment by JonRowe** [comment]
LGTM

**Pull request merged**